src/render/src/lib.rs

```diff
@@ -143,3 +148,80 @@ pub mod pso;
 pub mod shade;
 /// Convenience macros
 pub mod macros;
+*/
+
+use std::ops;
+use std::sync::mpsc;
+
+pub struct Gpu<B: Backend> {
+    device: Device<B>,
+    queue: CommandQueue<B, core::General>,
+    garbage: handle::GarbageReceiver<B>,
+}
+
+impl<B: Backend> Gpu<B> {
+    pub fn new<A: core::Adapter<B>>(adapter: &A) -> Self {
+        let queue_descs = adapter.get_queue_families()
+            .iter()
+            .map(|&(ref family, qtype)| (family, qtype, family.num_queues()) )
+            .collect::<Vec<_>>();
+        
```

Comment:
Somehow we need to filter for queues which can be used for presentation with the display surface.

Reply:
might be easier with #1468